### DISCUSSION TITLE
Speedrun Enhancements

### DISCUSSION THREAD

**Discussion by CRGorman** [scp-fs2open/fs2open.github.com#6131] [Ideas]
I don't have solid ideas to write issues for these yet, so discussion it is.

1. Clock: The idea for this one is once the first time the ready room is pressed, a timer starts counting up in real time and then doesn't stop until the end of the campaign is hit. Ideally, it would also give times for the end of a segment, or when a mission is set to complete. For a visual reference, this is what Livesplit looks like: 
![image](https://github.com/scp-fs2open/fs2open.github.com/assets/15680886/e91fc7e9-0874-45e1-9c18-68a8133683fa)
2. Shared pipe to Livesplit: Sorta like the previous idea, but a bit more complicated. Livesplit allows for using [pipes](https://learn.microsoft.com/en-us/windows/win32/ipc/pipes) to message the application some data. This I don't think can be fit into a mod and would likely require some code changes. The thing that stands out to me as being a problem is not knowing how this would work on Linux/Mac, as I have only done it on Windows.
3. Load Remover: This just pauses the clock whenever the loading screen is visible, nothing more.

This, coupled with #6075 and its PR #6128 might probably be best filed under a flag just labeled `-speedrun`.

Open to other suggestions/ideas.

**Comment by Goober5000** [comment]
For 1, there is a script in the fso-scripts repository that displays a nice large clock.  It could be used by another (new) script to display a clock that runs when you begin a mission and pauses when you exit a mission.
![screen0620](https://github.com/scp-fs2open/fs2open.github.com/assets/1878523/dd0e19ef-67ec-4472-9974-bf672df0fac5)
